cpp/pagerank_module/algorithm_online/pagerank.cpp

```diff
@@ -0,0 +1,222 @@
+
+    auto position = std::find(walk.begin(), walk.end(), from) + 1;
+
+    if (position == walk.end()) {
+      continue;
+    }
+
+    while (position != walk.end()) {
+      auto node_id = *position;
+      pagerank_online_alg::context.walks_table[node_id].erase(walk_index);
+      pagerank_online_alg::context.walks_counter[node_id]--;
+      position++;
+    }
+    walk.erase(std::find(walk.begin(), walk.end(), from) + 1, walk.end());
+
+    auto current_id = from;
+
+    // Skip creating routes if node does not exist anymore
+    if (!graph.NodeExists(current_id)) {
+      continue;
+    }
+
+    auto half_eps = pagerank_online_alg::global_epsilon / 2.0;
+    CreateRoute(graph, graph.GetInnerNodeId(current_id), walk, walk_index, half_eps,
+                *pagerank_online_alg::context.distr, *pagerank_online_alg::context.gen);
+  }
+}
+
+void UpdateDelete(const mg_graph::GraphView<> &graph, const std::uint64_t removed_vertex) {
+  pagerank_online_alg::context.walks_table.erase(removed_vertex);
+  pagerank_online_alg::context.walks_counter.erase(removed_vertex);
+}
+
+bool IsIncosistent(const mg_graph::GraphView<> &graph) {
+  for (auto const [node_id] : graph.Nodes()) {
+    auto external_id = graph.GetMemgraphNodeId(node_id);
+    if (pagerank_online_alg::context.walks_counter.find(external_id) ==
+        pagerank_online_alg::context.walks_counter.end()) {
+      return true;
+    }
+  }
+  return false;
+}
+}  // namespace
+
+std::vector<std::pair<std::uint64_t, double>> SetPagerank(const mg_graph::GraphView<> &graph, const std::uint64_t R,
+                                                          const double epsilon) {
+  pagerank_online_alg::global_R = R;
+  pagerank_online_alg::global_epsilon = epsilon;
+  pagerank_online_alg::context.Init();
+
+  auto walk_index = 0;
+  for (auto [node_id] : graph.Nodes()) {
+    // We have R random walks for each node in the graph
+    for (std::uint64_t i = 0; i < R; i++) {
+      std::vector<std::uint64_t> walk;
+
+      auto current_id = graph.GetMemgraphNodeId(node_id);
+      walk.emplace_back(current_id);
+      pagerank_online_alg::context.walks_table[current_id].insert(walk_index);
+      pagerank_online_alg::context.walks_counter[current_id]++;
+
+      CreateRoute(graph, graph.GetInnerNodeId(current_id), walk, walk_index, epsilon,
+                  *pagerank_online_alg::context.distr, *pagerank_online_alg::context.gen);
+
+      pagerank_online_alg::context.walks.emplace_back(std::move(walk));
+      walk_index++;
+    }
+  }
+
+  return CalculatePageRank();
+}
+
+std::vector<std::pair<std::uint64_t, double>> GetPagerank(const mg_graph::GraphView<> &graph) {
+  if (pagerank_online_alg::context.IsEmpty()) {
+    return SetPagerank(graph);
+  }
+  if (IsIncosistent(graph)) {
+    throw std::runtime_error(
+        "Graph has been modified, therefore is incosistent with cached results, please update the Pagerank by calling "
+        "set/reset!");
+  }
+  return CalculatePageRank();
```

Comment:
spaces between lines

Reply:
This is automatic formatting from clang-format